app/tests/tests_models.py

```diff
@@ -29,29 +29,13 @@ def test_verify_foreign_key_constraint_by_deleting_table_which_should_delete_col
 
     def test_verify_table_is_not_deleted_when_column_is_deleted(self):
         # Arrange
-        self.create_metadata_table_and_column()
+        create_metadata_table_and_column()
 
         # Act
         ColumnMetadata.objects.first().delete()
 
         # Assert
-        self.assertEqual(1, TableMetadata.objects.count(), "Table should not be deleted")
-        self.assertEqual(0, ColumnMetadata.objects.count(), "Column should be deleted")
-
-    @staticmethod
-    def create_metadata_table_and_column():
-        table_metadata = TableMetadata(name="TestTable",
-                                       original_file_name="test_file.csv",
-                                       description="Table created for testing")
-        table_metadata.save()
-
-        column_data_type = ColumnDataType(display_name="Text")
-        column_data_type.save()
-
-        column = ColumnMetadata(table_metadata=table_metadata, name="ColumnName",
-                                title="Friendly Column Name",
-                                description="Description of column",
-                                data_type=column_data_type,
-                                allow_missing_value=True,
-                                allow_duplicate_value=True)
-        column.save()
+        table_count = TableMetadata.objects.count()
```

Comment:
Add that we count whether that specific table exist and not just how many tables exists?

(not sure the suggestion is correct - this would also return `True` if table exists and not count)

```suggestion
        table_count = TableMetadata.objects.filter(table_id=1).exists()
```